addons/sale/sale.py

```diff
@@ -268,6 +268,13 @@ def create(self, vals):
         result = super(SaleOrder, self).create(vals)
         return result
 
+    @api.multi
+    def copy_data(self, default=None):
```

Comment:
Add **if default is None: default = {}**. Otherwise it will raise an error:  'NoneType' object does not support item assignment .